database/download.go

```diff
@@ -7,15 +7,24 @@ import (
 	"gitlab.com/NebulousLabs/errors"
 	"go.mongodb.org/mongo-driver/bson"
 	"go.mongodb.org/mongo-driver/bson/primitive"
-	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.mongodb.org/mongo-driver/mongo"
 )
 
 // Download describes a single download of a skylink by a user.
 type Download struct {
-	ID        primitive.ObjectID  `bson:"_id,omitempty" json:"_id"`
-	UserID    primitive.ObjectID  `bson:"user_id,omitempty" json:"user_id"`
-	SkylinkID primitive.ObjectID  `bson:"skylink_id,omitempty" json:"skylink_id"`
-	Timestamp primitive.Timestamp `bson:"timestamp" json:"timestamp"`
+	ID        primitive.ObjectID `bson:"_id,omitempty" json:"_id"`
+	UserID    primitive.ObjectID `bson:"user_id,omitempty" json:"user_id"`
+	SkylinkID primitive.ObjectID `bson:"skylink_id,omitempty" json:"skylink_id"`
+	Timestamp time.Time          `bson:"timestamp" json:"timestamp"`
+}
+
+// DownloadResponseDTO is the DTO we send as response to the caller.
+type DownloadResponseDTO struct {
+	ID        string    `bson:"string_id" json:"id"`
```

Comment:
why is this annotated with `bson`? This has nothing to do with mongo right?

Reply:
It does. When we fetch data from mongo we want to deserialize it into this DTO before we return it to the handler. We need to know how to map the fields, so `cursor.All` can populate that for us.

Is there a better way to do it?

Reply:
wasn't aware stuff got deserialized into these structs, 
figured you were populating them but I missed:
```
downloads := make([]DownloadResponseDTO, 0)
err = c.All(ctx, &downloads)		err = c.All(ctx, &downloads)
```